File: taskboard-backend-node/src/routes/tasks.js
const express = require('express');
const router = express.Router();

let tasks = [];
let nextId = 1;

router.get('/', (req, res) => {
    res.json(tasks);
});

router.post('/', (req, res) => {
    const { title, description } = req.body;
    if (!title) {
        return res.status(400).json({ error: 'Title is required' });
    }
    const newTask = { id: nextId++, title, description };
    tasks.push(newTask);
    res.status(201).json(newTask);
});

router.put('/:id', (req, res) => {
    const taskId = parseInt(req.params.id);
    const { title, description } = req.body;

    const task = tasks.find(t => t.id === taskId);
    if (!task) return res.status(404).json({ error: 'Task not found' });

    task.title = title ?? task.title;
    task.description = description ?? task.description;

    res.json(task);
});

router.delete('/:id', (req, res) => {
    const taskId = parseInt(req.params.id);
    const index = tasks.findIndex(t => t.id === taskId);
    if (index === -1) return res.status(404).json({ error: 'Task not found' });

    tasks.splice(index, 1);
    res.json({ message: 'Task deleted' });
});

module.exports = router;

File: backend-fastapi/app/api/v1/tasks.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import List, Optional

router = APIRouter()

TASKS = []

class Task(BaseModel):
    id: int
    title: str
    description: Optional[str] = None

class TaskCreate(BaseModel):
    title: str
    description: Optional[str] = None

@router.get("/", response_model=List[Task])
def get_tasks():
    return TASKS

@router.post("/", response_model=Task)
def create_task(task: TaskCreate):
    new_task = Task(id=len(TASKS)+1, title=task.title, description=task.description)
    TASKS.append(new_task)
    return new_task

@router.put("/{task_id}", response_model=Task)
def update_task(task_id: int, task: TaskCreate):
    for t in TASKS:
        if t.id == task_id:
            t.title = task.title
            t.description = task.description
            return t
    raise HTTPException(status_code=404, detail="Task not found")

@router.delete("/{task_id}")
def delete_task(task_id: int):
    for t in TASKS:
        if t.id == task_id:
            TASKS.remove(t)
            return {"message": "Task deleted"}
    raise HTTPException(status_code=404, detail="Task not found")
